smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+# TUTORIAL: Smart Contract usage with the Javascript SDK
+
+This tutorial explains how to use the æternity javascrip SDK to create smart contracts and interact with them in the æternity network.
+
+## Prerequisites
+
+ - up to date npm and nodejs installed
+ - private and public key for an account with some balance ([create an account](account-creation-in-ae-cli.md); [get testnet tokens](get-testnet-tokens.md))
+
+## Setup
+
+1. Create a directory for the project and initialize it with npm:
+
+```
+mkdir aepp-smart-contract-usage-tutorial
+cd aepp-smart-contract-usage-tutorial
+npm init -y
+```
+
+2. Install the aepp-sdk dependency `npm install --save @aeternity/aepp-sdk`
+
+3. Create a example project file `index.js` in this directory
+
+4. Setup example code structure in `index.js`, this does require the SDK dependency and setup a sync function in which we can build the naming workflow
+```
+const fs = require('fs');
+const {Universal} = require('@aeternity/aepp-sdk');
+
+// function to hold all our example code
+const main = async (name) => {
+    // insert all the following here
+};
+```
+
+5. Defining helper functions to later catch possible errors in a developer friendly was, add after the `const main = async (name) => {` line:
+
+```
+// function to log any errors
+const logError = (error) => console.error(error);
+
+// function to decode errors from contract calling
+const decodeError = async (error) => {
+
+    // results from the node need to be decoded to be human readable
+    const decodedError = await client.contractDecodeData('string', error.returnValue).catch(logError);
+    console.error('error:', decodedError.value);
+};
+```
+
+## Define Example Contract
```

Comment:
Maybe we can place this section before the creation of index.js in order to get the smart contract "out of the way". This way we would not require the reader to "context switch" from js/sdk to Sophia and back to js/sdk